scripts/aggregate_cells.py

```diff
@@ -0,0 +1,72 @@
+import os
+import pandas as pd
+from pycytominer.cyto_utils.cells import SingleCells
+
+
+def aggregate(sql_file, metadata_dir, barcode_platemap):
+    """aggregates single cell data into aggregate profiles
+
+    Paramters:
+    ----------
+    sql_file: str
+            SQl file that contains single cell data obtain from a single plate
+    metadata_dir : str 
+        associated metadata file with the single cell data
+    barcode_platemap : str 
+        file containing the barcode id of each platedata
+
+    Returns:
+    --------
+        No object is returned
+
+        Generates cell count, aggregate aggregate, augmented aggregate profiles, and normalized
+        augmented aggregate profiles in the results/ directory.
+    """
+
+    # Loading appropriate platemaps with given plate data
+    plate = os.path.basename(sql_file).rsplit(".", 1)
+    barcode_platemap_df = pd.read_csv(barcode_platemap)
+    platemap = barcode_platemap_df.query(
+        "Assay_Plate_Barcode == @plate"
+    ).Plate_Map_Name.values[0]
+    platemap_file = os.path.join(metadata_dir, "platemap", "{}.csv".format(platemap))
+    platemap_df = pd.read_csv(platemap_file)
+
+    sqlite_file = "sqlite:///{}".format(sql_file)
+    strata = ["Image_Metadata_Plate", "Image_Metadata_Well"]
+    image_cols = ["TableNumber", "ImageNumber"]
+    single_cell_profile = SingleCells(
```

Comment:
These hard coded parameters will be removed once we agree on an appropriate structure how the configuration files should be.

Below is an example to give you an idea how configurational files work in `snakemake` 

Let say that this is your `config.yaml` file and it contains all the parameters that are known in `SingleCell`. 
```yaml
#config.yaml
  Parameters:
    strata: ["Metadata_Plate", "Metadata_Well"],
    aggregation_operation: "median",
    output_file: "none",
    compartments: default_compartments,
    compartment_linking_cols: "default_linking_cols",
    merge_cols: ["TableNumber", "ImageNumber"],
    image_cols: ["TableNumber", "ImageNumber", "Metadata_Site"],
    add_image_features: False,
    image_feature_categories: None,
    features: "infer",
    load_image_data: True,
    subsample_frac: 1,
    subsample_n: "all",
    subsampling_random_state: "none",
    fields_of_view: "all",
    fields_of_view_feature: "Metadata_Site",
    object_feature: "Metadata_ObjectNumber"
```
In the `.smk` or `Snakemake`  file, can load the configuration file by using `configfile:` attribute and then  call the configurations under the `params` attribute:

```python
# in test.smk
configfile: "/path/to/config/file.yaml"

rule aggregate:
    input:
        "path/to/input"
    output:
        "path/to/ouput/files"
    params:
        single_cell=config["single_cell"]["paramters"]
    script:
        "test.py"
``` 
Snakemake will inject the configurational parameters as a dictionary! This provides a great advantages because we can apply `**kwags` parameter into the `SingleCell`!

```python
# in test.py
from pycytominer.cyto_utils.cells import SingleCells


# Name hashing comes from the .smk and not from the config file
single_cell_params = snakemake.params["single_cell"] # ->dict {"strata": ["Metadata_Plate", "Metadata_Well"] ...}

# unpacking dictionary and using key
sp = SingleCells(**single_cell_params)


```
However, consistency is key! The parameter names in the `configuration.yaml` class must be identical to `SingleCell`'s parameter names The `snakemake.params'  will return a dictionary where the keys are the parameters names and the values are the parameter values. That's how `SingleCell`  class knows what parameters are being used! 

here's a small example using a dictionary to fill in parameters in a class:
```python
# class with required parameters
class TestClass:
    def __init__(self, data, method, save):
        self.data = data
        self.method = method
        self.save = save

# usage example
# variable t is identical to variable x
t = TestClass(data="path/to/data", method="simplify", save=True)

# params is the type of data we would expect from snakemake.params
params = {"data": "path/to/data", "method": "simplify", "save": True}
x = TestClass(**params)

```
I personally think this is really helpful  because the user does not have to go into the code to know which parameters are being used. they can just read the config file. 

Makes it easier for us developers because we don't have two write all these parameters explicitly in the source code :) 

Hopefully this answers your question!

Reply:
Beautifully! I'm going to transfer this description to the discussion in #3